### PR TITLE
Fixed issue in flooding algorithm

### PR DESCRIPTION
**Pull request by ErikBjare**
I discovered a pretty severe bug that caused >1h of activity to not show up on the dashboard.

I narrowed it down to an issue in the afk bucket:

image

The logs complained upon query:

    [WARNING]: Gap was of negative duration (-4815.292s). This error will only show once per batch.

The two first events begin at the exact same time, which causes the negative gap to occur when the following is computed `gap = e2.timestamp - (e1.timestamp + e1.duration)`. The problem arose when the flooding were to take place, since it happened *despite* the negative gap being detected, which it should not, leading to shortening of the longer event in my case to the end of the shorter event.

I also realized that if two events had a negative gap and contained the same data, they can be safely merged into the same event.

### CODE REVIEW

**Comment by johan-bjareholt** [comment]
Looks fine to me.

Also, feel free to review my flood implementation in aw-server-rust, I had to rewrite it quite a bit due to Rust didn't allow iterating over the same list twice with mutability (which is what zip(events, events) does)

**Comment by ErikBjare** [comment]
Also @johan-bjareholt you might have noticed that I removed a bunch of debug-logging from query2. This will actually have some performance increases, as `"{}".format(large_object)` which happened quite frequently runs a dict-printer despite logging level being set below `DEBUG`.

File: aw_transform/flood.py [hunk 1]
```diff
@@ -15,38 +15,42 @@ def flood(events: List[Event], pulsetime: float=5) -> List[Event]:
...
+            if not warned_about_negative_gap_safe:
+                logger.warning("Gap was of negative duration but could be safely merged ({}s). This message will only show once per batch.".format(gap.total_seconds()))
+                warned_about_negative_gap_safe = True
+        elif gap < -negative_gap_trim_thres and not warned_about_negative_gap_unsafe:
+            # Events with negative gap but differing data cannot be merged safely
+            logger.warning("Gap was of negative duration and could NOT be safely merged ({}s). This warning will only show once per batch.".format(gap.total_seconds()))
+            warned_about_negative_gap_unsafe = True
+            # logger.warning("Event 1 (id {}): {} {}".format(e1.id, e1.timestamp, e1.duration))
+            # logger.warning("Event 2 (id {}): {} {}".format(e2.id, e2.timestamp, e2.duration))
+        elif -negative_gap_trim_thres < gap <= timedelta(seconds=pulsetime):
```
Comment (johan-bjareholt) [inline aw_transform/flood.py]: I don't get it, why is negative_gap_trim_thres needed?
Is timedelta a float?

Comment (ErikBjare) [reply aw_transform/flood.py]: Not really, afaik it represented with an int for seconds and an int for microseconds.

But I noticed there were a few -0.001s gaps appearing in the logs which should be harmless to trim.

**Comment by johan-bjareholt** [comment]
@ErikBjare And about the prints in query2, most of them should be removed completely (I thought I had, but apparently not).
Printing to DEBUG is IMO pretty worthless for debugging since so much other stuff is often spewed out from other unrelated code so I never use that.

**Comment by ErikBjare** [comment]
Merging now

**Pull request merged**